**Portfolio assignment 8**

15 min:

- Calculate the 90%, 95%, 99% and 99.99% confidence interval for at least 2 columns with numerical data in the dataset that you chose in portfolio assignment 4. Do you see the impact the confidence has on the interval?
- Now calculate the 95% confidence interval again but use only the first 10% of your rows. Compare this interval to the previous 95% confidence interval you calculated. Do you see the impact of having less data?

In [4]:
import pandas as pd
import scipy.stats as st

In [2]:
stockx = pd.read_excel('StockX_Dataset.xlsx')

In [3]:
stockx.head()

Order Date   Brand                                   Sneaker Name  \
0 2017-09-01   Yeezy           Adidas-Yeezy-Boost-350-Low-V2-Beluga   
1 2017-09-01   Yeezy    Adidas-Yeezy-Boost-350-V2-Core-Black-Copper   
2 2017-09-01   Yeezy     Adidas-Yeezy-Boost-350-V2-Core-Black-Green   
3 2017-09-01   Yeezy       Adidas-Yeezy-Boost-350-V2-Core-Black-Red   
4 2017-09-01   Yeezy  Adidas-Yeezy-Boost-350-V2-Core-Black-Red-2017   

   Sale Price  Retail Price Release Date  Shoe Size  Buyer Region  
0      1097.0           220   2016-09-24       11.0    California  
1       685.0           220   2016-11-23       11.0    California  
2       690.0           220   2016-11-23       11.0    California  
3      1075.0           220   2016-11-23       11.5      Kentucky  
4       828.0           220   2017-02-11       11.0  Rhode Island

### Sale Price

In [5]:
confidence = 0.90
st.t.interval(confidence, len(stockx)-1, loc=stockx['Sale Price'].mean(), scale=st.sem(stockx['Sale Price']))

(445.30295787866123, 447.96655065909533)

In [7]:
confidence = 0.95
st.t.interval(confidence, len(stockx)-1, loc=stockx['Sale Price'].mean(), scale=st.sem(stockx['Sale Price']))

(445.0478164813223, 448.2216920564343)

In [8]:
confidence = 0.99
st.t.interval(confidence, len(stockx)-1, loc=stockx['Sale Price'].mean(), scale=st.sem(stockx['Sale Price']))

(444.5491499167756, 448.72035862098096)

In [9]:
confidence = 0.999
st.t.interval(confidence, len(stockx)-1, loc=stockx['Sale Price'].mean(), scale=st.sem(stockx['Sale Price']))

(443.97044385454535, 449.29906468321127)

### Shoe Size

In [11]:
confidence = 0.90
st.t.interval(confidence, len(stockx)-1, loc=stockx['Shoe Size'].mean(), scale=st.sem(stockx['Shoe Size']))

(9.332061348454824, 9.35630153121223)

In [12]:
confidence = 0.95
st.t.interval(confidence, len(stockx)-1, loc=stockx['Shoe Size'].mean(), scale=st.sem(stockx['Shoe Size']))

(9.329739419162214, 9.35862346050484)

In [13]:
confidence = 0.99
st.t.interval(confidence, len(stockx)-1, loc=stockx['Shoe Size'].mean(), scale=st.sem(stockx['Shoe Size']))

(9.32520127476251, 9.363161604904544)

In [14]:
confidence = 0.99
st.t.interval(confidence, len(stockx)-1, loc=stockx['Shoe Size'].mean(), scale=st.sem(stockx['Shoe Size']))

(9.32520127476251, 9.363161604904544)

### 10% Dataset

In [17]:
# 10% Dataset
ten_percent = len(stockx) * 0.1
data = stockx.head(int(ten_percent))

In [33]:
confidence = 0.95
st.t.interval(confidence, len(data) - 1, loc=data['Sale Price'].mean(), scale=st.sem(data['Sale Price']))

(534.4853096991403, 551.3636147630907)

In [20]:
confidence = 0.95
st.t.interval(confidence, len(data) - 1, loc=data['Shoe Size'].mean(), scale=st.sem(data['Shoe Size']))

(9.580977195742882, 9.66704681626312)

Ik zie dat er best wel grote verschillen onstaan:
- Sale Price: (445.0478164813223, 448.2216920564343) --- (537.898657676093, 547.9502667861381)
- Shoe Size: (9.329739419162214, 9.35862346050484) --- (9.580977195742882, 9.66704681626312)

Als de dataset dus kleiner wordt zal het confidence interval groter worden.